marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/OutputEndpointImpl.java

```diff
@@ -79,14 +87,17 @@ public void awaitCompletion() {
                     throw new RuntimeException("error while calling "+getEndpoint().getEndpointPath(), throwable);
                 }
 
+                InputStream[] result = null;
+                if(output != null) {
+                    result = output.toArray(size -> new InputStream[size]);
+                }
                 if (allowsEndpointState()) {
-                    if(output!= null && output.findFirst()!= null && output.findFirst().isPresent()) {
-                        InputStream[] result = output.toArray(size -> new InputStream[size]);
-                        setEndpointState(result[0]);
-                        for(int i=1; i<result.length; i++) {
-                            forEachOutput((Consumer<InputStream>) result[i]);
+                        if(result !=null) {
```

Comment:
Should there be a test that the result.length is greater than 0?  Instead of nested if branches, would the code be easier to read as an && condition on the test of allowsEndpointState()?